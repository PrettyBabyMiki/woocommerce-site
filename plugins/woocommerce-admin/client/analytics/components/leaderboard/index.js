/** @format */
/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { Component } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { get } from 'lodash';
import PropTypes from 'prop-types';

/**
 * WooCommerce dependencies
 */
import { Card, EmptyTable, TableCard } from '@woocommerce/components';

/**
 * Internal dependencies
 */
import ReportError from 'analytics/components/report-error';
import { getReportTableData } from 'store/reports/utils';
import withSelect from 'wc-api/with-select';
import './style.scss';

export class Leaderboard extends Component {
	render() {
		const {
			getHeadersContent,
			getRowsContent,
			isRequesting,
			isError,
			items,
			tableQuery,
			title,
		} = this.props;
		const data = get( items, [ 'data' ], [] );
		const rows = getRowsContent( data );
		const totalRows = tableQuery ? tableQuery.per_page : 5;

		if ( isError ) {
			return <ReportError className="woocommerce-leaderboard" isError />;
		}

		if ( ! isRequesting && rows.length === 0 ) {
			return (
				<Card title={ title } className="woocommerce-leaderboard">
					<EmptyTable>
						{ __( 'No data recorded for the selected time period.', 'wc-admin' ) }
					</EmptyTable>
				</Card>
			);
		}

		return (
			<TableCard
				className="woocommerce-leaderboard"
				headers={ getHeadersContent() }
				isLoading={ isRequesting }
				rows={ rows }
				rowsPerPage={ totalRows }
				showMenu={ false }
				title={ title }
				totalRows={ totalRows }
			/>
		);
	}
}

Leaderboard.propTypes = {
	/**
	 * The endpoint to use in API calls to populate the table rows and summary.
	 * For example, if `taxes` is provided, data will be fetched from the report
	 * `taxes` endpoint (ie: `/wc/v3/reports/taxes` and `/wc/v3/reports/taxes/stats`).
	 * If the provided endpoint doesn't exist, an error will be shown to the user
	 * with `ReportError`.
	 */
	endpoint: PropTypes.string,
	/**
	 * A function that returns the headers object to build the table.
	 */
	getHeadersContent: PropTypes.func.isRequired,
	/**
	 * A function that returns the rows array to build the table.
	 */
	getRowsContent: PropTypes.func.isRequired,
	/**
	 * Query args added to the report table endpoint request.
	 */
	query: PropTypes.object,
	/**
	 * Properties to be added to the query sent to the report table endpoint.
	 */
	tableQuery: PropTypes.object,
	/**
	 * String to display as the title of the table.
	 */
	title: PropTypes.string.isRequired,
};

export default compose(
	withSelect( ( select, props ) => {
		const { endpoint, tableQuery, query } = props;
		const tableData = getReportTableData( endpoint, query, select, tableQuery );

		return { ...tableData };
	} )
)( Leaderboard );
