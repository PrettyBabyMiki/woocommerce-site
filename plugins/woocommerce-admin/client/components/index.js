/** @format */

export { default as Chart } from './chart';
export { default as ChartLegend } from './chart/legend';
export { default as ChartPlaceholder } from './chart/placeholder';
export { default as D3Chart } from './chart/charts';
