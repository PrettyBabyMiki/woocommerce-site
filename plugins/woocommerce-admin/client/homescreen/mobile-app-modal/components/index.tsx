export { JetpackInstallationStepper } from './JetpackInstallationStepper';
export {
	useJetpackPluginState,
	JetpackPluginStates,
} from './useJetpackPluginState';
export { useSendMagicLink } from './useSendMagicLink';
export { SendMagicLinkButton } from './SendMagicLinkButton';
