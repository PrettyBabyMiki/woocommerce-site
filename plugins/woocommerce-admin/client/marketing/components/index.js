export { default as Button } from './button';
export { default as Card } from './card';
export { default as ProductIcon } from './product-icon';
export { default as Slider } from './slider';
