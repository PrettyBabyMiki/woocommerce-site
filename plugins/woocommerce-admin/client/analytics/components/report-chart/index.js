/** @format */
/**
 * External dependencies
 */
import { Component } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { format as formatDate } from '@wordpress/date';
import { withSelect } from '@wordpress/data';
import PropTypes from 'prop-types';
import { find, get } from 'lodash';

/**
 * WooCommerce dependencies
 */
import { flattenFilters } from '@woocommerce/navigation';
import {
	getAllowedIntervalsForQuery,
	getCurrentDates,
	getDateFormatsForInterval,
	getIntervalForQuery,
	getChartTypeForQuery,
	getPreviousDate,
} from '@woocommerce/date';
import { Chart } from '@woocommerce/components';

/**
 * Internal dependencies
 */
import { getReportChartData, getTooltipValueFormat } from 'store/reports/utils';
import ReportError from 'analytics/components/report-error';

export const DEFAULT_FILTER = 'all';

export class ReportChart extends Component {
	getSelectedFilter( filters, query ) {
		if ( filters.length === 0 ) {
			return null;
		}

		const filterConfig = filters.pop();

		if ( filterConfig.showFilters( query ) ) {
			const allFilters = flattenFilters( filterConfig.filters );
			const value = query[ filterConfig.param ] || DEFAULT_FILTER;
			const selectedFilter = find( allFilters, { value } );
			const selectedFilterParam = get( selectedFilter, [ 'settings', 'param' ] );

			if ( ! selectedFilterParam || Object.keys( query ).includes( selectedFilterParam ) ) {
				return selectedFilter;
			}
		}

		return this.getSelectedFilter( filters, query );
	}

	getChartMode() {
		const { filters, query } = this.props;
		if ( ! filters ) {
			return;
		}
		const clonedFilters = filters.slice( 0 );
		const selectedFilter = this.getSelectedFilter( clonedFilters, query );

		return get( selectedFilter, [ 'chartMode' ] );
	}

	render() {
		const { query, itemsLabel, path, primaryData, secondaryData, selectedChart } = this.props;

		if ( primaryData.isError || secondaryData.isError ) {
			return <ReportError isError />;
		}

		const currentInterval = getIntervalForQuery( query );
		const allowedIntervals = getAllowedIntervalsForQuery( query );
		const formats = getDateFormatsForInterval( currentInterval, primaryData.data.intervals.length );
		const { primary, secondary } = getCurrentDates( query );
		const primaryKey = `${ primary.label } (${ primary.range })`;
		const secondaryKey = `${ secondary.label } (${ secondary.range })`;

		const chartData = primaryData.data.intervals.map( function( interval, index ) {
			const secondaryDate = getPreviousDate(
				formatDate( 'Y-m-d', interval.date_start ),
				primary.after,
				secondary.after,
				query.compare,
				currentInterval
			);

			const secondaryInterval = secondaryData.data.intervals[ index ];
			return {
				date: formatDate( 'Y-m-d\\TH:i:s', interval.date_start ),
				[ primaryKey ]: {
					labelDate: interval.date_start,
					value: interval.subtotals[ selectedChart.key ] || 0,
				},
				[ secondaryKey ]: {
					labelDate: secondaryDate.format( 'YYYY-MM-DD HH:mm:ss' ),
					value: ( secondaryInterval && secondaryInterval.subtotals[ selectedChart.key ] ) || 0,
				},
			};
		} );

		return (
			<Chart
				path={ path }
				query={ query }
				data={ chartData }
				title={ selectedChart.label }
				interval={ currentInterval }
				type={ getChartTypeForQuery( query ) }
				allowedIntervals={ allowedIntervals }
				itemsLabel={ itemsLabel }
				mode={ this.getChartMode() }
				tooltipLabelFormat={ formats.tooltipLabelFormat }
				tooltipValueFormat={ getTooltipValueFormat( selectedChart.type ) }
				tooltipTitle={ selectedChart.label }
				xFormat={ formats.xFormat }
				x2Format={ formats.x2Format }
				dateParser={ '%Y-%m-%dT%H:%M:%S' }
				valueType={ selectedChart.type }
				isRequesting={ primaryData.isRequesting || secondaryData.isRequesting }
			/>
		);
	}
}

ReportChart.propTypes = {
	filters: PropTypes.array,
	itemsLabel: PropTypes.string,
	path: PropTypes.string.isRequired,
	primaryData: PropTypes.object.isRequired,
	query: PropTypes.object.isRequired,
	secondaryData: PropTypes.object.isRequired,
	selectedChart: PropTypes.object.isRequired,
};

export default compose(
	withSelect( ( select, props ) => {
		const { query, endpoint } = props;
		const primaryData = getReportChartData( endpoint, 'primary', query, select );
		const secondaryData = getReportChartData( endpoint, 'secondary', query, select );
		return {
			primaryData,
			secondaryData,
		};
	} )
)( ReportChart );
