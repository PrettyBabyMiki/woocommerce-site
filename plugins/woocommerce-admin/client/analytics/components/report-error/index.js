/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { Component } from '@wordpress/element';
import PropTypes from 'prop-types';
import { EmptyContent } from '@woocommerce/components';
import { getAdminLink } from '@woocommerce/wc-admin-settings';

/**
 * Component to render when there is an error in a report component due to data
 * not being loaded or being invalid.
 */
class ReportError extends Component {
	render() {
		const { className, isError, isEmpty } = this.props;
		let title, actionLabel, actionURL, actionCallback;

		if ( isError ) {
			title = __(
				'There was an error getting your stats. Please try again.',
				'woocommerce-admin'
			);
			actionLabel = __( 'Reload', 'woocommerce-admin' );
			actionCallback = () => {
				// @todo Add tracking for how often an error is displayed, and the reload action is clicked.
				window.location.reload();
			};
		} else if ( isEmpty ) {
			title = __(
				'No results could be found for this date range.',
				'woocommerce-admin'
			);
			actionLabel = __( 'View Orders', 'woocommerce-admin' );
			actionURL = getAdminLink( 'edit.php?post_type=shop_order' );
		}
		return (
			<EmptyContent
				className={ className }
				title={ title }
				actionLabel={ actionLabel }
				actionURL={ actionURL }
				actionCallback={ actionCallback }
			/>
		);
	}
}

ReportError.propTypes = {
	/**
	 * Additional class name to style the component.
	 */
	className: PropTypes.string,
	/**
	 * Boolean representing whether there was an error.
	 */
	isError: PropTypes.bool,
	/**
	 * Boolean representing whether the issue is that there is no data.
	 */
	isEmpty: PropTypes.bool,
};

ReportError.defaultProps = {
	className: '',
};

export default ReportError;
