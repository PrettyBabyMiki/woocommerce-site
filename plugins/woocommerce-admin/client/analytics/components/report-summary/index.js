/** @format */
/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { Component } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { map } from 'lodash';
import { withSelect } from '@wordpress/data';
import PropTypes from 'prop-types';

/**
 * WooCommerce dependencies
 */
import { formatCurrency } from '@woocommerce/currency';
import { getDateParamsFromQuery } from '@woocommerce/date';
import { getNewPath } from '@woocommerce/navigation';
import { SummaryList, SummaryListPlaceholder, SummaryNumber } from '@woocommerce/components';

/**
 * Internal dependencies
 */
import { getSummaryNumbers } from 'store/reports/utils';
import { numberFormat } from 'lib/number';
import ReportError from 'analytics/components/report-error';

class ReportSummary extends Component {
	render() {
		const { selectedChart, charts } = this.props;

		if ( this.props.summaryNumbers.isError ) {
			return <ReportError isError />;
		}

		if ( this.props.summaryNumbers.isRequesting ) {
			return <SummaryListPlaceholder numberOfItems={ charts.length } />;
		}

		const totals = this.props.summaryNumbers.totals.primary || {};
		const secondaryTotals = this.props.summaryNumbers.totals.secondary || {};
		const { compare } = getDateParamsFromQuery( this.props.query );

		const summaryNumbers = map( charts, chart => {
			const { key, label, type } = chart;
			const isSelected = selectedChart.key === key;
			let value = parseFloat( totals[ key ] );
			let secondaryValue =
				( secondaryTotals[ key ] && parseFloat( secondaryTotals[ key ] ) ) || undefined;

			let delta = 0;
			if ( secondaryValue && secondaryValue !== 0 ) {
				delta = Math.round( ( value - secondaryValue ) / secondaryValue * 100 );
			}

			switch ( type ) {
				case 'average':
					value = Math.round( value );
					secondaryValue = secondaryValue && Math.round( secondaryValue );
					break;
				case 'currency':
					value = formatCurrency( value );
					secondaryValue = secondaryValue && formatCurrency( secondaryValue );
					break;
				case 'number':
					value = numberFormat( value );
					secondaryValue = secondaryValue && numberFormat( secondaryValue );
					break;
			}

			const href = getNewPath( { chart: key } );

			return (
				<SummaryNumber
					key={ key }
					value={ value }
					label={ label }
					selected={ isSelected }
					prevValue={ secondaryValue }
					prevLabel={
						'previous_period' === compare
							? __( 'Previous Period:', 'wc-admin' )
							: __( 'Previous Year:', 'wc-admin' )
					}
					delta={ delta }
					href={ href }
				/>
			);
		} );

		return <SummaryList>{ summaryNumbers }</SummaryList>;
	}
}

ReportSummary.propTypes = {
	charts: PropTypes.array.isRequired,
	endpoint: PropTypes.string.isRequired,
	query: PropTypes.object.isRequired,
	selectedChart: PropTypes.object.isRequired,
};

export default compose(
	withSelect( ( select, props ) => {
		const { query, endpoint } = props;
		const summaryNumbers = getSummaryNumbers( endpoint, query, select );

		return {
			summaryNumbers,
		};
	} )
)( ReportSummary );
