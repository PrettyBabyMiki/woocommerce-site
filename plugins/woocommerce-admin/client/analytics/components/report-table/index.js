/** @format */
/**
 * External dependencies
 */
import { applyFilters } from '@wordpress/hooks';
import { Component, createRef, Fragment } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { focus } from '@wordpress/dom';
import { withDispatch } from '@wordpress/data';
import { get } from 'lodash';
import PropTypes from 'prop-types';

/**
 * WooCommerce dependencies
 */
import { TableCard } from '@woocommerce/components';
import { onQueryChange } from '@woocommerce/navigation';

/**
 * Internal dependencies
 */
import ReportError from 'analytics/components/report-error';
import { getReportChartData, getReportTableData } from 'wc-api/reports/utils';
import { QUERY_DEFAULTS } from 'wc-api/constants';
import withSelect from 'wc-api/with-select';
import { extendTableData } from './utils';

import './style.scss';

const TABLE_FILTER = 'woocommerce_admin_report_table';

/**
 * Component that extends `TableCard` to facilitate its usage in reports.
 */
class ReportTable extends Component {
	constructor( props ) {
		super( props );

		this.onColumnsChange = this.onColumnsChange.bind( this );
		this.onPageChange = this.onPageChange.bind( this );
		this.scrollPointRef = createRef();
	}

	onColumnsChange( shownColumns ) {
		const { columnPrefsKey, getHeadersContent, updateCurrentUserData } = this.props;
		const columns = getHeadersContent().map( header => header.key );
		const hiddenColumns = columns.filter( column => ! shownColumns.includes( column ) );

		if ( columnPrefsKey ) {
			const userDataFields = {
				[ columnPrefsKey ]: hiddenColumns,
			};
			updateCurrentUserData( userDataFields );
		}
	}

	onPageChange() {
		this.scrollPointRef.current.scrollIntoView();
		const tableElement = this.scrollPointRef.current.nextSibling.querySelector(
			'.woocommerce-table__table'
		);
		const focusableElements = focus.focusable.find( tableElement );

		if ( focusableElements.length ) {
			focusableElements[ 0 ].focus();
		}
	}

	filterShownHeaders( headers, hiddenKeys ) {
		if ( ! hiddenKeys || ! hiddenKeys.length ) {
			return headers;
		}

		return headers.map( header => {
			const hidden = hiddenKeys.includes( header.key ) && ! header.required;
			return { ...header, hiddenByDefault: hidden };
		} );
	}

	render() {
		const {
			getHeadersContent,
			getRowsContent,
			getSummary,
			isRequesting,
			itemIdField,
			primaryData,
			tableData,
			// These props are not used in the render function, but are destructured
			// so they are not included in the `tableProps` variable.
			endpoint,
			tableQuery,
			userPrefColumns,
			...tableProps
		} = this.props;

		const { items, query } = tableData;

		const isError = tableData.isError || primaryData.isError;

		if ( isError ) {
			return <ReportError isError />;
		}

		const isLoading = isRequesting || tableData.isRequesting || primaryData.isRequesting;
		const totals = get( primaryData, [ 'data', 'totals' ], {} );
		const totalResults = items.totalResults;
		const downloadable = 0 < totalResults;
		const { headers, ids, rows, summary } = applyFilters( TABLE_FILTER, {
			endpoint: endpoint,
			headers: getHeadersContent(),
			ids: itemIdField ? items.data.map( item => item[ itemIdField ] ) : [],
			rows: getRowsContent( items.data ),
			totals: totals,
			summary: getSummary ? getSummary( totals, totalResults ) : null,
		} );

		// Hide any headers based on user prefs, if loaded.
		const filteredHeaders = this.filterShownHeaders( headers, userPrefColumns );

		return (
			<Fragment>
				<div
					className="woocommerce-report-table__scroll-point"
					ref={ this.scrollPointRef }
					aria-hidden
				/>
				<TableCard
					downloadable={ downloadable }
					headers={ filteredHeaders }
					ids={ ids }
					isLoading={ isLoading }
					onQueryChange={ onQueryChange }
					onColumnsChange={ this.onColumnsChange }
					onPageChange={ this.onPageChange }
					rows={ rows }
					rowsPerPage={ parseInt( query.per_page ) || QUERY_DEFAULTS.pageSize }
					summary={ summary }
					totalRows={ totalResults }
					{ ...tableProps }
				/>
			</Fragment>
		);
	}
}

ReportTable.propTypes = {
	/**
	 * The key for user preferences settings for column visibility.
	 */
	columnPrefsKey: PropTypes.string,
	/**
	 * The endpoint to use in API calls to populate the table rows and summary.
	 * For example, if `taxes` is provided, data will be fetched from the report
	 * `taxes` endpoint (ie: `/wc/v4/reports/taxes` and `/wc/v4/reports/taxes/stats`).
	 * If the provided endpoint doesn't exist, an error will be shown to the user
	 * with `ReportError`.
	 */
	endpoint: PropTypes.string,
	/**
	 * Name of the methods available via `select( 'wc-api' )` that will be used to
	 * load more data for table items. If omitted, no call will be made and only
	 * the data returned by the reports endpoint will be used.
	 */
	extendItemsMethodNames: PropTypes.shape( {
		getError: PropTypes.string,
		isRequesting: PropTypes.string,
		load: PropTypes.string,
	} ),
	/**
	 * A function that returns the headers object to build the table.
	 */
	getHeadersContent: PropTypes.func.isRequired,
	/**
	 * A function that returns the rows array to build the table.
	 */
	getRowsContent: PropTypes.func.isRequired,
	/**
	 * A function that returns the summary object to build the table.
	 */
	getSummary: PropTypes.func,
	/**
	 * The name of the property in the item object which contains the id.
	 */
	itemIdField: PropTypes.string,
	/**
	 * Primary data of that report. If it's not provided, it will be automatically
	 * loaded via the provided `endpoint`.
	 */
	primaryData: PropTypes.object,
	/**
	 * Table data of that report. If it's not provided, it will be automatically
	 * loaded via the provided `endpoint`.
	 */
	tableData: PropTypes.object.isRequired,
	/**
	 * Properties to be added to the query sent to the report table endpoint.
	 */
	tableQuery: PropTypes.object,
	/**
	 * String to display as the title of the table.
	 */
	title: PropTypes.string.isRequired,
};

ReportTable.defaultProps = {
	primaryData: {},
	tableData: {
		items: {
			data: [],
			totalResults: 0,
		},
		query: {},
	},
	tableQuery: {},
};

export default compose(
	withSelect( ( select, props ) => {
		const {
			endpoint,
			getSummary,
			isRequesting,
			query,
			tableData,
			tableQuery,
			columnPrefsKey,
			filters,
			advancedFilters,
		} = props;

		let userPrefColumns = [];
		if ( columnPrefsKey ) {
			const { getCurrentUserData } = select( 'wc-api' );
			const userData = getCurrentUserData();

			userPrefColumns = userData[ columnPrefsKey ];
		}

		if ( isRequesting || ( query.search && ! ( query[ endpoint ] && query[ endpoint ].length ) ) ) {
			return {
				userPrefColumns,
			};
		}
		// Variations and Category charts are powered by the /reports/products/stats endpoint.
		const chartEndpoint = [ 'variations', 'categories' ].includes( endpoint )
			? 'products'
			: endpoint;
		const primaryData = getSummary
			? getReportChartData( {
					endpoint: chartEndpoint,
					dataType: 'primary',
					query,
					select,
					filters,
					advancedFilters,
					tableQuery,
				} )
			: {};
		const queriedTableData =
			tableData ||
			getReportTableData( { endpoint, query, select, tableQuery, filters, advancedFilters } );
		const extendedTableData = extendTableData( select, props, queriedTableData );

		return {
			primaryData,
			tableData: extendedTableData,
			query: { ...tableQuery, ...query },
			userPrefColumns,
		};
	} ),
	withDispatch( dispatch => {
		const { updateCurrentUserData } = dispatch( 'wc-api' );

		return {
			updateCurrentUserData,
		};
	} )
)( ReportTable );
