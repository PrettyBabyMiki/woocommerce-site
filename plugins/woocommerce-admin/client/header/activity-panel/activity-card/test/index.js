/**
 * External dependencies
 */
import { Button } from '@wordpress/components';
import Gridicon from 'gridicons';
import { shallow } from 'enzyme';
import moment from 'moment';
import { Gravatar } from '@woocommerce/components';

/**
 * Internal dependencies
 */
import { ActivityCard } from '../';

describe( 'ActivityCard', () => {
	test( 'should have correct title', () => {
		const card = (
			<ActivityCard title="Inbox message">
				This card has some content
			</ActivityCard>
		);
		expect( card.props.title ).toBe( 'Inbox message' );
	} );

	test( 'should render a basic card', () => {
		const card = shallow(
			<ActivityCard title="Inbox message">
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render an unread bubble on a card', () => {
		const card = shallow(
			<ActivityCard title="Inbox message" unread>
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a custom icon on a card', () => {
		const card = shallow(
			<ActivityCard
				title="Inbox message"
				icon={ <Gridicon icon="customize" /> }
			>
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a gravatar on a card', () => {
		const card = shallow(
			<ActivityCard
				title="Inbox message"
				icon={ <Gravatar user="admin@local.test" /> }
			>
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a timestamp on a card', () => {
		// We're generating this via moment to ensure it's always "3 days ago".
		const threeDaysAgo = moment().subtract( 3, 'days' ).format();
		const card = shallow(
			<ActivityCard title="Inbox message" date={ threeDaysAgo }>
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render an action on a card', () => {
		const noop = () => {};
		const card = shallow(
			<ActivityCard
				title="Inbox message"
				actions={
					<Button isSecondary onClick={ noop }>
						Action
					</Button>
				}
			>
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render multiple actions on a card', () => {
		const noop = () => {};
		const card = shallow(
			<ActivityCard
				title="Inbox message"
				actions={ [
					<Button key="action1" isPrimary onClick={ noop }>
						Action 1
					</Button>,
					<Button key="action2" isSecondary onClick={ noop }>
						Action 2
					</Button>,
				] }
			>
				This card has some content
			</ActivityCard>
		);
		expect( card ).toMatchSnapshot();
	} );
} );
