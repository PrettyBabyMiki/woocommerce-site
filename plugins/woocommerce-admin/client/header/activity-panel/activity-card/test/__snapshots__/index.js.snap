// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard should render a basic card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <t
      icon="notice-outline"
      size={48}
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
</section>
`;

exports[`ActivityCard should render a custom icon on a card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <t
      icon="customize"
      size={24}
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
</section>
`;

exports[`ActivityCard should render a gravatar on a card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <Gravatar
      size={60}
      user="admin@local.test"
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
</section>
`;

exports[`ActivityCard should render a timestamp on a card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <t
      icon="notice-outline"
      size={48}
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
    <span
      className="woocommerce-activity-card__date"
    >
      3 days ago
    </span>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
</section>
`;

exports[`ActivityCard should render an action on a card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <t
      icon="notice-outline"
      size={48}
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
  <footer
    className="woocommerce-activity-card__actions"
  >
    <ForwardRef(Button)
      isSecondary={true}
      key="0"
      onClick={[Function]}
    >
      Action
    </ForwardRef(Button)>
  </footer>
</section>
`;

exports[`ActivityCard should render an unread bubble on a card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    className="woocommerce-activity-card__unread"
  />
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <t
      icon="notice-outline"
      size={48}
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
</section>
`;

exports[`ActivityCard should render multiple actions on a card 1`] = `
<section
  className="woocommerce-activity-card"
>
  <span
    aria-hidden={true}
    className="woocommerce-activity-card__icon"
  >
    <t
      icon="notice-outline"
      size={48}
    />
  </span>
  <header
    className="woocommerce-activity-card__header"
  >
    <H
      className="woocommerce-activity-card__title"
    >
      Inbox message
    </H>
  </header>
  <Section
    className="woocommerce-activity-card__body"
  >
    This card has some content
  </Section>
  <footer
    className="woocommerce-activity-card__actions"
  >
    <ForwardRef(Button)
      isPrimary={true}
      key="0"
      onClick={[Function]}
    >
      Action 1
    </ForwardRef(Button)>
    <ForwardRef(Button)
      isSecondary={true}
      key="1"
      onClick={[Function]}
    >
      Action 2
    </ForwardRef(Button)>
  </footer>
</section>
`;
